concordium-consensus/src/Concordium/KonsensusV1/Types.hs

```diff
@@ -1226,24 +1250,73 @@ instance HashableTo BlockQuasiHash (BakedBlock pv) where
               where
                 transactionsAndOutcomesHash = Hash.hashOfHashes transactionsHash outcomesHash
                   where
-                    transactionsHash = computeTransactionsHash bbTransactions
+                    transactionsHash = computeTransactionsHash (sing @bhv) bbTransactions
                     outcomesHash = tohGet dbhv0TransactionOutcomesHash
                 stateHash = v0StateHash dbhv0BlockStateHash
             DerivableBlockHashesV1{..} -> Hash.hashOfHashes transactionsHash blockResultHash
               where
-                transactionsHash = computeTransactionsHash bbTransactions
+                transactionsHash = computeTransactionsHash (sing @bhv) bbTransactions
                 blockResultHash = theBlockResultHash dbhv1BlockResultHash
 
+instance (IsProtocolVersion pv) => HashableTo SuccessorProof (BakedBlock pv) where
+    getHash = makeSuccessorProof @(BlockHashVersionFor pv) . getHash
+
 -- | Compute the block hash from the header hash and quasi-hash.
-computeBlockHash :: BlockHeaderHash -> BlockQuasiHash -> BlockHash
-computeBlockHash bhh bqh =
+computeBlockHash' :: BlockHeaderHash -> Hash.Hash -> BlockHash
+{-# INLINE computeBlockHash' #-}
+computeBlockHash' bhh bqh =
     BlockHash $
         Hash.hashOfHashes
             (theBlockHeaderHash bhh)
-            (theBlockQuasiHash bqh)
+            bqh
 
-instance HashableTo BlockHash (BakedBlock pv) where
-    getHash bb = computeBlockHash (getHash bb) (getHash bb)
+-- | Compute the block hash from the header hash and quasi-hash.
+computeBlockHash :: BlockHeaderHash -> BlockQuasiHash' bhv -> BlockHash
+computeBlockHash bhh bqh = computeBlockHash' bhh (theBlockQuasiHash bqh)
+
+instance (IsProtocolVersion pv) => HashableTo BlockHash (BakedBlock pv) where
+    getHash bb = computeBlockHash @(BlockHashVersionFor pv) (getHash bb) (getHash bb)
+
+instance (Monad m, BlockHashVersionFor pv ~ 'BlockHashVersion1) => Merkle.MerkleProvable m (BakedBlock pv) where
```

Comment:
I might have missed them, but I can't seem to find any unit tests for this implementation.

I would expect a worked example or two coded as a unit test so we have something stable to compare things to when making changes.

Reply:
Added tests.